### PR TITLE
Builder Plugin Support

### PR DESCRIPTION
**Pull request by Cryrivers**

### CODE REVIEW

File: packages/plugins/plugin-mock-example/README.md [hunk 1]
```diff
@@ -19,7 +20,7 @@ userName: string;
  * @example 10
  * @precision 1
  */
- score: number
+score: number;
 ```
 
-- Will generate data based on example
+- Will generate data based on example, muliple examples will be chosen randomly.
```
Comment (tanhauhau) [inline packages/plugins/plugin-mock-example/README.md]: oops. this change can commit directly, separately from this MR

File: packages/core/core/docs/BuilderPlugin.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  name: string;
+  supportedExtensions: string[];
+  buildConfigFile(
+    configFilePath: string,
+    destDir: string,
+    verbose: boolean = false,
+    importHelpers: boolean = true,
+  ): Promise<string>;
+  buildSourceCode(sourceCode: string): Promise<string>;
+}
```
Comment (tanhauhau) [inline packages/core/core/docs/BuilderPlugin.md]: different from the definition in `plugin/index.ts`

File: packages/core/core/src/plugin/index.ts [hunk 3]
```diff
@@ -25,7 +29,13 @@ export interface MockPlugin {
...
   supportedExtensions: string[];
-  build(configFilePath: string): Promise<string[]>;
+  build(
+    configFilePath: string,
+    destDir: string,
+    verbose?: boolean,
+    importHelpers?: boolean,
+  ): Promise<string>;
+  transpile(sourceCode: string): Promise<string>;
 }
```
Comment (tanhauhau) [inline packages/core/core/src/plugin/index.ts]: why do we need `build` and `transpile`, what is the difference?
why both return `Promise<string>`?

Comment (Cryrivers) [reply packages/core/core/src/plugin/index.ts]: build compiles a file, return the file path to compiled file

transpile compiles source code in-memory and returns transpiled code, which is used in testing

Comment (Cryrivers) [reply packages/core/core/src/plugin/index.ts]: updated

**Pull request merged**